src/media-segment-request.js

```diff
@@ -102,31 +96,29 @@ const getProgressStats = (progressEvent) => {
  * @param {Object} request -  the XHR request that possibly generated the error
  */
 const handleErrors = (error, request) => {
-  const response = request.response;
-
-  if (!request.aborted && error) {
+  if (request.timedout) {
```

Comment:
Can you go into detail of what the issue was and why changing the order of checking for timeout then errors solves it?